metrics/ci_integrations/docs/03_ci_integration_verbose_mode.md

```diff
@@ -0,0 +1,152 @@
+# CI Integrations Verbose Mode
+
+## TL;DR
+
+Introducing a `verbose mode` for the [`CI Integrations`](https://github.com/platform-platform/monorepo/tree/master/metrics/ci_integrations) tool. Running the `CI Integrations` tool in the `verbose mode` gives more context and provides additional details of what it's doing while working.
+
+## References
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [CI Integrations Tool Architecture](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/01_ci_integration_module_architecture.md)
+
+## Goals
+> Identify success metrics and measurable goals.
+
+- A clear design of the CI Integrations verbose mode.
+- An overview of steps the verbose mode implementation requires.
+
+## Design
+> Explain and diagram the technical design.
+
+The following section provides an implementation of the `verbose mode` and changes that need to do to the existing `CI Integrations` codebase.
+
+Here is a list of changes we should provide:
+1. [Change the existing Logger class](#change-the-logger).
+2. [Remove accepting the Logger instance](#remove-accepting-the-logger-instance).
+3. [Replace logging messages and errors](#replace-logging-messages-and-errors).
+4. [Add verbose option](#add-verbose-option).
+5. [Setup the Logger](#setup-logger).
+6. [Logging actions](#logging-actions).
+
+### Change the Logger
+
+To make logging actions easily, we need to convert the existing `Logger` class to the class with static methods. It allows us to show logs in different parts of the `CI Integrations` tool without the need to accept the logger instance in every class, that wants to log actions.
+
+Here is an example of how the `Logger` class should look:
+
+```
+class Logger {
+  static IOSink _errorSink = stderr;
+
+  static IOSink _messageSink = stdout;
+
+  static bool _verbose = false;
+
+  static void setup({
+    IOSink errorSink,
+    IOSink messageSink,
+    bool verbose,
+  }) {
+    _errorSink = errorSink ?? stderr;
+    _messageSink = messageSink ?? stdout;
+    _verbose = verbose ?? false;
+  }
+
+  static void printError(Object error) => _errorSink.writeln(error);
+
+  static void printMessage(Object message) => _messageSink.writeln(message);
+
+  static void printLog(Object message) {
```

Comment:
Would it be great to change the name of the `Logger` methods to make the API more clear, like `Logger.logInfo`?

```suggestion
  static void logInfo(Object message) {
```